### ISSUE TITLE
Faililing tests integration tests on Windows

### ISSUE THREAD

**Issue by piotr-iohk** [cardano-foundation/cardano-wallet#1115]
# Context
 
#703

# Test Case

Following 21 tests are failing (after the path issue fix -> #1109):
```
Failures:

  test/integration/Cardano/Wallet/Jormungandr/NetworkSpec.hs:135:9:
  1) No backend required, Cardano.Wallet.NetworkSpec, Happy Paths, getAccountState: successfully gets account state
       uncaught exception: IOException of type PermissionDenied
       C:\Users\katar\AppData\Local\Temp\cardano-wallet-jormungandr-dbc61946bb87b0ae\chain\blocks.sqlite-shm: removeDirectoryRecursive:removeContentsRecursive:removePathRecursive:removeContentsRecursive:removePathRecursive:DeleteFile "\\\\?\\C:\\Users\\katar\\AppData\\Local\\Temp\\cardano-wallet-jormungandr-dbc61946bb87b0ae\\chain\\blocks.sqlite-shm": permission denied (Odmowa dostępu.)

  To rerun use: --match "/No backend required/Cardano.Wallet.NetworkSpec/Happy Paths/getAccountState: successfully gets account state/"

  test/integration/Cardano/Wallet/Jormungandr/NetworkSpec.hs:147:9:
  2) No backend required, Cardano.Wallet.NetworkSpec, Happy Paths, get network tip
       uncaught exception: IOException of type PermissionDenied
       C:\Users\katar\AppData\Local\Temp\cardano-wallet-jormungandr-fddede207e198264\chain\blocks.sqlite-shm: removeDirectoryRecursive:removeContentsRecursive:removePathRecursive:removeContentsRecursive:removePathRecursive:DeleteFile "\\\\?\\C:\\Users\\katar\\AppData\\Local\\Temp\\cardano-wallet-jormungandr-fddede207e198264\\chain\\blocks.sqlite-shm": permission denied (Odmowa dostępu.)

  To rerun use: --match "/No backend required/Cardano.Wallet.NetworkSpec/Happy Paths/get network tip/"

  test/integration/Cardano/Wallet/Jormungandr/NetworkSpec.hs:155:9:
  3) No backend required, Cardano.Wallet.NetworkSpec, Happy Paths, get some blocks from the genesis
       uncaught exception: IOException of type PermissionDenied
       C:\Users\katar\AppData\Local\Temp\cardano-wallet-jormungandr-4fddc9cc3517e081\chain\blocks.sqlite-shm: removeDirectoryRecursive:removeContentsRecursive:removePathRecursive:removeContentsRecursive:removePathRecursive:DeleteFile "\\\\?\\C:\\Users\\katar\\AppData\\Local\\Temp\\cardano-wallet-jormungandr-4fddc9cc3517e081\\chain\\blocks.sqlite-shm": permission denied (Odmowa dostępu.)

  To rerun use: --match "/No backend required/Cardano.Wallet.NetworkSpec/Happy Paths/get some blocks from the genesis/"

  test/integration/Cardano/Wallet/Jormungandr/NetworkSpec.hs:161:9:
  4) No backend required, Cardano.Wallet.NetworkSpec, Happy Paths, no blocks after the tip
       uncaught exception: IOException of type PermissionDenied
       C:\Users\katar\AppData\Local\Temp\cardano-wallet-jormungandr-b366f54b64748f5c\chain\blocks.sqlite-shm: removeDirectoryRecursive:removeContentsRecursive:removePathRecursive:removeContentsRecursive:removePathRecursive:DeleteFile "\\\\?\\C:\\Users\\katar\\AppData\\Local\\Temp\\cardano-wallet-jormungandr-b366f54b64748f5c\\chain\\blocks.sqlite-shm": permission denied (Odmowa dostępu.)

  To rerun use: --match "/No backend required/Cardano.Wallet.NetworkSpec/Happy Paths/no blocks after the tip/"

  test/integration/Cardano/Wallet/Jormungandr/NetworkSpec.hs:175:9:
  5) No backend required, Cardano.Wallet.NetworkSpec, Happy Paths, initiates recovery when the intersection is unknown
       uncaught exception: IOException of type PermissionDenied
       C:\Users\katar\AppData\Local\Temp\cardano-wallet-jormungandr-1f48d76bd7205e5a\chain\blocks.sqlite-shm: removeDirectoryRecursive:removeContentsRecursive:removePathRecursive:removeContentsRecursive:removePathRecursive:DeleteFile "\\\\?\\C:\\Users\\katar\\AppData\\Local\\Temp\\cardano-wallet-jormungandr-1f48d76bd7205e5a\\chain\\blocks.sqlite-shm": permission denied (Odmowa dostępu.)

  To rerun use: --match "/No backend required/Cardano.Wallet.NetworkSpec/Happy Paths/initiates recovery when the intersection is unknown/"

  test/integration/Cardano/Wallet/Jormungandr/NetworkSpec.hs:207:9:
  6) No backend required, Cardano.Wallet.NetworkSpec, Error paths, networkTip: ErrNetworkUnreachable
       uncaught exception: IOException of type OtherError
       Network.Socket.connect: <socket: 652>: failed (Connection refused (WSAECONNREFUSED))

  To rerun use: --match "/No backend required/Cardano.Wallet.NetworkSpec/Error paths/networkTip: ErrNetworkUnreachable/"

  test/integration/Cardano/Wallet/Jormungandr/NetworkSpec.hs:221:9:
  7) No backend required, Cardano.Wallet.NetworkSpec, Error paths, nextBlocks: ErrNetworkUnreachable
       uncaught exception: IOException of type OtherError
       Network.Socket.connect: <socket: 524>: failed (Connection refused (WSAECONNREFUSED))

  To rerun use: --match "/No backend required/Cardano.Wallet.NetworkSpec/Error paths/nextBlocks: ErrNetworkUnreachable/"

  test/integration/Cardano/Wallet/Jormungandr/NetworkSpec.hs:235:9:
  8) No backend required, Cardano.Wallet.NetworkSpec, Error paths, networkTip: throws on invalid url
       uncaught exception: IOException of type PermissionDenied
       C:\Users\katar\AppData\Local\Temp\cardano-wallet-jormungandr-67fb685c5e01c5dd\chain\blocks.sqlite-shm: removeDirectoryRecursive:removeContentsRecursive:removePathRecursive:removeContentsRecursive:removePathRecursive:DeleteFile "\\\\?\\C:\\Users\\katar\\AppData\\Local\\Temp\\cardano-wallet-jormungandr-67fb685c5e01c5dd\\chain\\blocks.sqlite-shm": permission denied (Odmowa dostępu.)

  To rerun use: --match "/No backend required/Cardano.Wallet.NetworkSpec/Error paths/networkTip: throws on invalid url/"

  test/integration/Cardano/Wallet/Jormungandr/NetworkSpec.hs:407:5:
  9) No backend required, Cardano.Wallet.NetworkSpec, White-box error path tests, getAccountState, resource not found
       uncaught exception: IOException of type PermissionDenied
       C:\Users\katar\AppData\Local\Temp\cardano-wallet-jormungandr-3ad4ffd19d43325e\chain\blocks.sqlite-shm: removeDirectoryRecursive:removeContentsRecursive:removePathRecursive:removeContentsRecursive:removePathRecursive:DeleteFile "\\\\?\\C:\\Users\\katar\\AppData\\Local\\Temp\\cardano-wallet-jormungandr-3ad4ffd19d43325e\\chain\\blocks.sqlite-shm": permission denied (Odmowa dostępu.)

  To rerun use: --match "/No backend required/Cardano.Wallet.NetworkSpec/White-box error path tests/getAccountState/resource not found/"

  test/integration/Cardano/Wallet/Jormungandr/NetworkSpec.hs:418:5:
  10) No backend required, Cardano.Wallet.NetworkSpec, White-box error path tests, getAccountState, invalid url path
       uncaught exception: IOException of type PermissionDenied
       C:\Users\katar\AppData\Local\Temp\cardano-wallet-jormungandr-28c4b532bf1095aa\chain\blocks.sqlite-shm: removeDirectoryRecursive:removeContentsRecursive:removePathRecursive:removeContentsRecursive:removePathRecursive:DeleteFile "\\\\?\\C:\\Users\\katar\\AppData\\Local\\Temp\\cardano-wallet-jormungandr-28c4b532bf1095aa\\chain\\blocks.sqlite-shm": permission denied (Odmowa dostępu.)

  To rerun use: --match "/No backend required/Cardano.Wallet.NetworkSpec/White-box error path tests/getAccountState/invalid url path/"

  test/integration/Cardano/Wallet/Jormungandr/NetworkSpec.hs:407:5:
  11) No backend required, Cardano.Wallet.NetworkSpec, White-box error path tests, getBlock, resource not found
       uncaught exception: IOException of type PermissionDenied
       C:\Users\katar\AppData\Local\Temp\cardano-wallet-jormungandr-64a38408f72f7499\chain\blocks.sqlite-shm: removeDirectoryRecursive:removeContentsRecursive:removePathRecursive:removeContentsRecursive:removePathRecursive:DeleteFile "\\\\?\\C:\\Users\\katar\\AppData\\Local\\Temp\\cardano-wallet-jormungandr-64a38408f72f7499\\chain\\blocks.sqlite-shm": permission denied (Odmowa dostępu.)

  To rerun use: --match "/No backend required/Cardano.Wallet.NetworkSpec/White-box error path tests/getBlock/resource not found/"

  test/integration/Cardano/Wallet/Jormungandr/NetworkSpec.hs:418:5:
  12) No backend required, Cardano.Wallet.NetworkSpec, White-box error path tests, getBlock, invalid url path
       uncaught exception: IOException of type PermissionDenied
       C:\Users\katar\AppData\Local\Temp\cardano-wallet-jormungandr-26baa8c5033b9330\chain\blocks.sqlite-shm: removeDirectoryRecursive:removeContentsRecursive:removePathRecursive:removeContentsRecursive:removePathRecursive:DeleteFile "\\\\?\\C:\\Users\\katar\\AppData\\Local\\Temp\\cardano-wallet-jormungandr-26baa8c5033b9330\\chain\\blocks.sqlite-shm": permission denied (Odmowa dostępu.)

  To rerun use: --match "/No backend required/Cardano.Wallet.NetworkSpec/White-box error path tests/getBlock/invalid url path/"

  test/integration/Cardano/Wallet/Jormungandr/NetworkSpec.hs:418:5:
  13) No backend required, Cardano.Wallet.NetworkSpec, White-box error path tests, getDescendantIds, invalid url path
       uncaught exception: IOException of type PermissionDenied
       C:\Users\katar\AppData\Local\Temp\cardano-wallet-jormungandr-ce69fae002d81ebf\chain\blocks.sqlite-shm: removeDirectoryRecursive:removeContentsRecursive:removePathRecursive:removeContentsRecursive:removePathRecursive:DeleteFile "\\\\?\\C:\\Users\\katar\\AppData\\Local\\Temp\\cardano-wallet-jormungandr-ce69fae002d81ebf\\chain\\blocks.sqlite-shm": permission denied (Odmowa dostępu.)

  To rerun use: --match "/No backend required/Cardano.Wallet.NetworkSpec/White-box error path tests/getDescendantIds/invalid url path/"

  test/integration/Cardano/Wallet/Jormungandr/NetworkSpec.hs:418:5:
  14) No backend required, Cardano.Wallet.NetworkSpec, White-box error path tests, getStakeDistribution, invalid url path
       uncaught exception: IOException of type PermissionDenied
       C:\Users\katar\AppData\Local\Temp\cardano-wallet-jormungandr-30830b20ed38229d\chain\blocks.sqlite-shm: removeDirectoryRecursive:removeContentsRecursive:removePathRecursive:removeContentsRecursive:removePathRecursive:DeleteFile "\\\\?\\C:\\Users\\katar\\AppData\\Local\\Temp\\cardano-wallet-jormungandr-30830b20ed38229d\\chain\\blocks.sqlite-shm": permission denied (Odmowa dostępu.)

  To rerun use: --match "/No backend required/Cardano.Wallet.NetworkSpec/White-box error path tests/getStakeDistribution/invalid url path/"

  test/integration/Cardano/Wallet/Jormungandr/NetworkSpec.hs:343:9:
  15) No backend required, Cardano.Wallet.NetworkSpec, Submitting signed transactions (that are not obviously wrong), empty tx succeeds
       uncaught exception: IOException of type PermissionDenied
       C:\Users\katar\AppData\Local\Temp\cardano-wallet-jormungandr-9fa856d3e650e7be\chain\blocks.sqlite-shm: removeDirectoryRecursive:removeContentsRecursive:removePathRecursive:removeContentsRecursive:removePathRecursive:DeleteFile "\\\\?\\C:\\Users\\katar\\AppData\\Local\\Temp\\cardano-wallet-jormungandr-9fa856d3e650e7be\\chain\\blocks.sqlite-shm": permission denied (Odmowa dostępu.)

  To rerun use: --match "/No backend required/Cardano.Wallet.NetworkSpec/Submitting signed transactions (that are not obviously wrong)/empty tx succeeds/"

  src/Test/Integration/Scenario/CLI/Miscellaneous.hs:35:9:
  16) No backend required, Miscellaneous CLI tests, CLI_VERSION - cardano-wallet shows version
       expected: "2019.11.18 (git revision: 0000000000000000000000000000000000000000)"
        but got: "2019.11.18 (git revision: 47c2103e33efeaa72509785ac3c9971d137eb1c8)"

  To rerun use: --match "/No backend required/Miscellaneous CLI tests/CLI_VERSION - cardano-wallet shows version/"

  src/Test/Integration/Framework/DSL.hs:608:36:
  17) API Specifications TRANS_EXTERNAL_CREATE_01 - proper single output transaction and proper binary format
       expected: 1
        but got: 2

  To rerun use: --match "/API Specifications/TRANS_EXTERNAL_CREATE_01 - proper single output transaction and proper binary format/"

  test/integration/Test/Integration/Jormungandr/Scenario/CLI/Server.hs:100:13:
  18) CLI Specifications, DaedalusIPC [SERIAL], Should reply with the port --random
       expected: ExitSuccess
        but got: ExitFailure 1

  To rerun use: --match "/CLI Specifications/DaedalusIPC [SERIAL]/Should reply with the port --random/"

  test/integration/Test/Integration/Jormungandr/Scenario/CLI/Server.hs:102:9:
  19) CLI Specifications, DaedalusIPC [SERIAL], Should reply with the port --port
       uncaught exception: IOException of type OtherError
       Network.Socket.connect: <socket: 856>: failed (Connection refused (WSAECONNREFUSED))

  To rerun use: --match "/CLI Specifications/DaedalusIPC [SERIAL]/Should reply with the port --port/"

  test/integration/Test/Integration/Jormungandr/Scenario/CLI/Server.hs:113:13:
  20) CLI Specifications, DaedalusIPC [SERIAL], Regression test for #1036
       expected: ExitSuccess
        but got: ExitFailure 1

  To rerun use: --match "/CLI Specifications/DaedalusIPC [SERIAL]/Regression test for #1036/"

  test/integration/Test/Integration/Jormungandr/Scenario/CLI/Server.hs:169:17:
  21) CLI Specifications, LOGGING - cardano-wallet serve logging [SERIAL], LOGGING - Serve shuts down logging correctly
       "\ESC[34m[iohk.cardano-wallet:Info:ThreadId 3]\ESC[0m [2019-12-04 09:42:42.27 UTC] Creating directory: /does-not-exist\n\ESC[34m[iohk.cardano-wallet:Info:ThreadId 3]\ESC[0m [2019-12-04 09:42:42.27 UTC] Running as v2019.11.18 (git revision: 47c2103e33efeaa72509785ac3c9971d137eb1c8)\n\ESC[34m[iohk.cardano-wallet:Info:ThreadId 3]\ESC[0m [2019-12-04 09:42:42.27 UTC] Wallet backend server starting...\n\ESC[34m[iohk.cardano-wallet:Info:ThreadId 3]\ESC[0m [2019-12-04 09:42:42.27 UTC] Node is J\246rmungandr on testnet\n\ESC[34m[iohk.cardano-wallet:Info:ThreadId 23]\ESC[0m [2019-12-04 09:42:42.28 UTC] Waiting for J\246rmungandr to be ready on tcp/65072\n\ESC[34m[iohk.cardano-wallet.daedalus-ipc:Info:ThreadId 22]\ESC[0m [2019-12-04 09:42:42.27 UTC] Daedalus IPC is not enabled.\n\ESC[34m[iohk.cardano-wallet:Info:ThreadId 23]\ESC[0m [2019-12-04 09:42:42.28 UTC] J\246rmungandr is ready.\n[iohk.cardano-wallet:Debug:ThreadId 23] [2019-12-04 09:42:42.29 UTC] Using connection string: /does-not-exist\\stake-pools.sqlite\n[iohk.cardano-wallet.query:Debug:ThreadId 23] [2019-12-04 09:42:42.30 UTC] CREATE TABLE \"pool_production\"(\"pool_id\" VARCHAR NOT NULL,\"slot\" INTEGER NOT NULL,\"header_hash\" VARCHAR NOT NULL,\"parent_header_hash\" VARCHAR NOT NULL,\"block_height\" INTEGER NOT NULL, PRIMARY KEY (\"slot\"))\n[iohk.cardano-wallet.query:Debug:ThreadId 23] [2019-12-04 09:42:42.30 UTC] CREATE TABLE \"stake_distribution\"(\"pool_id\" VARCHAR NOT NULL,\"epoch\" INTEGER NOT NULL,\"stake\" INTEGER NOT NULL, PRIMARY KEY (\"pool_id\",\"epoch\"))\n\ESC[35m[iohk.cardano-wallet:Notice:ThreadId 23]\ESC[0m [2019-12-04 09:42:42.30 UTC] 2 migrations were applied to the database.\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:42.30 UTC] SELECT \"pool_id\", \"slot\", \"header_hash\", \"parent_header_hash\", \"block_height\" FROM \"pool_production\" ORDER BY \"slot\" DESC LIMIT 4\n\ESC[34m[iohk.cardano-wallet.stake-pools:Info:ThreadId 37]\ESC[0m [2019-12-04 09:42:42.31 UTC] Start monitoring stake pools. Currently at genesis\n\ESC[34m[iohk.cardano-wallet:Info:ThreadId 23]\ESC[0m [2019-12-04 09:42:42.31 UTC] Wallet backend server listening on 127.0.0.1:65165\n\ESC[34m[iohk.cardano-wallet.stake-pools:Info:ThreadId 37]\ESC[0m [2019-12-04 09:42:42.32 UTC] Applying blocks [481260.5 ... 481262.12]\n\ESC[34m[iohk.cardano-wallet.stake-pools:Info:ThreadId 37]\ESC[0m [2019-12-04 09:42:42.32 UTC] Writing stake-distribution for epoch 481262\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:42.32 UTC] DELETE FROM \"stake_distribution\" WHERE (\"epoch\"=?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:42.32 UTC] INSERT INTO \"stake_distribution\"(\"pool_id\",\"epoch\",\"stake\") VALUES (?,?,?),(?,?,?),(?,?,?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:42.32 UTC] INSERT INTO \"pool_production\"(\"pool_id\",\"slot\",\"header_hash\",\"parent_header_hash\",\"block_height\") VALUES(?,?,?,?,?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:42.32 UTC] INSERT INTO \"pool_production\"(\"pool_id\",\"slot\",\"header_hash\",\"parent_header_hash\",\"block_height\") VALUES(?,?,?,?,?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:42.32 UTC] INSERT INTO \"pool_production\"(\"pool_id\",\"slot\",\"header_hash\",\"parent_header_hash\",\"block_height\") VALUES(?,?,?,?,?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:42.32 UTC] INSERT INTO \"pool_production\"(\"pool_id\",\"slot\",\"header_hash\",\"parent_header_hash\",\"block_height\") VALUES(?,?,?,?,?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:42.32 UTC] INSERT INTO \"pool_production\"(\"pool_id\",\"slot\",\"header_hash\",\"parent_header_hash\",\"block_height\") VALUES(?,?,?,?,?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:42.32 UTC] INSERT INTO \"pool_production\"(\"pool_id\",\"slot\",\"header_hash\",\"parent_header_hash\",\"block_height\") VALUES(?,?,?,?,?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:42.32 UTC] INSERT INTO \"pool_production\"(\"pool_id\",\"slot\",\"header_hash\",\"parent_header_hash\",\"block_height\") VALUES(?,?,?,?,?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:42.33 UTC] INSERT INTO \"pool_production\"(\"pool_id\",\"slot\",\"header_hash\",\"parent_header_hash\",\"block_height\") VALUES(?,?,?,?,?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:42.33 UTC] INSERT INTO \"pool_production\"(\"pool_id\",\"slot\",\"header_hash\",\"parent_header_hash\",\"block_height\") VALUES(?,?,?,?,?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:42.33 UTC] INSERT INTO \"pool_production\"(\"pool_id\",\"slot\",\"header_hash\",\"parent_header_hash\",\"block_height\") VALUES(?,?,?,?,?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:42.33 UTC] INSERT INTO \"pool_production\"(\"pool_id\",\"slot\",\"header_hash\",\"parent_header_hash\",\"block_height\") VALUES(?,?,?,?,?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:42.33 UTC] INSERT INTO \"pool_production\"(\"pool_id\",\"slot\",\"header_hash\",\"parent_header_hash\",\"block_height\") VALUES(?,?,?,?,?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:42.33 UTC] INSERT INTO \"pool_production\"(\"pool_id\",\"slot\",\"header_hash\",\"parent_header_hash\",\"block_height\") VALUES(?,?,?,?,?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:42.33 UTC] INSERT INTO \"pool_production\"(\"pool_id\",\"slot\",\"header_hash\",\"parent_header_hash\",\"block_height\") VALUES(?,?,?,?,?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:42.33 UTC] INSERT INTO \"pool_production\"(\"pool_id\",\"slot\",\"header_hash\",\"parent_header_hash\",\"block_height\") VALUES(?,?,?,?,?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:42.33 UTC] INSERT INTO \"pool_production\"(\"pool_id\",\"slot\",\"header_hash\",\"parent_header_hash\",\"block_height\") VALUES(?,?,?,?,?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:42.33 UTC] INSERT INTO \"pool_production\"(\"pool_id\",\"slot\",\"header_hash\",\"parent_header_hash\",\"block_height\") VALUES(?,?,?,?,?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:42.33 UTC] INSERT INTO \"pool_production\"(\"pool_id\",\"slot\",\"header_hash\",\"parent_header_hash\",\"block_height\") VALUES(?,?,?,?,?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:42.33 UTC] INSERT INTO \"pool_production\"(\"pool_id\",\"slot\",\"header_hash\",\"parent_header_hash\",\"block_height\") VALUES(?,?,?,?,?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:42.33 UTC] INSERT INTO \"pool_production\"(\"pool_id\",\"slot\",\"header_hash\",\"parent_header_hash\",\"block_height\") VALUES(?,?,?,?,?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:42.33 UTC] INSERT INTO \"pool_production\"(\"pool_id\",\"slot\",\"header_hash\",\"parent_header_hash\",\"block_height\") VALUES(?,?,?,?,?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:42.33 UTC] INSERT INTO \"pool_production\"(\"pool_id\",\"slot\",\"header_hash\",\"parent_header_hash\",\"block_height\") VALUES(?,?,?,?,?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:42.33 UTC] INSERT INTO \"pool_production\"(\"pool_id\",\"slot\",\"header_hash\",\"parent_header_hash\",\"block_height\") VALUES(?,?,?,?,?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:42.33 UTC] INSERT INTO \"pool_production\"(\"pool_id\",\"slot\",\"header_hash\",\"parent_header_hash\",\"block_height\") VALUES(?,?,?,?,?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:42.33 UTC] INSERT INTO \"pool_production\"(\"pool_id\",\"slot\",\"header_hash\",\"parent_header_hash\",\"block_height\") VALUES(?,?,?,?,?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:42.33 UTC] INSERT INTO \"pool_production\"(\"pool_id\",\"slot\",\"header_hash\",\"parent_header_hash\",\"block_height\") VALUES(?,?,?,?,?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:42.33 UTC] INSERT INTO \"pool_production\"(\"pool_id\",\"slot\",\"header_hash\",\"parent_header_hash\",\"block_height\") VALUES(?,?,?,?,?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:42.33 UTC] INSERT INTO \"pool_production\"(\"pool_id\",\"slot\",\"header_hash\",\"parent_header_hash\",\"block_height\") VALUES(?,?,?,?,?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:42.33 UTC] INSERT INTO \"pool_production\"(\"pool_id\",\"slot\",\"header_hash\",\"parent_header_hash\",\"block_height\") VALUES(?,?,?,?,?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:42.33 UTC] INSERT INTO \"pool_production\"(\"pool_id\",\"slot\",\"header_hash\",\"parent_header_hash\",\"block_height\") VALUES(?,?,?,?,?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:42.33 UTC] INSERT INTO \"pool_production\"(\"pool_id\",\"slot\",\"header_hash\",\"parent_header_hash\",\"block_height\") VALUES(?,?,?,?,?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:42.33 UTC] INSERT INTO \"pool_production\"(\"pool_id\",\"slot\",\"header_hash\",\"parent_header_hash\",\"block_height\") VALUES(?,?,?,?,?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:42.33 UTC] INSERT INTO \"pool_production\"(\"pool_id\",\"slot\",\"header_hash\",\"parent_header_hash\",\"block_height\") VALUES(?,?,?,?,?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:42.33 UTC] INSERT INTO \"pool_production\"(\"pool_id\",\"slot\",\"header_hash\",\"parent_header_hash\",\"block_height\") VALUES(?,?,?,?,?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:42.33 UTC] INSERT INTO \"pool_production\"(\"pool_id\",\"slot\",\"header_hash\",\"parent_header_hash\",\"block_height\") VALUES(?,?,?,?,?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:42.33 UTC] INSERT INTO \"pool_production\"(\"pool_id\",\"slot\",\"header_hash\",\"parent_header_hash\",\"block_height\") VALUES(?,?,?,?,?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:42.33 UTC] INSERT INTO \"pool_production\"(\"pool_id\",\"slot\",\"header_hash\",\"parent_header_hash\",\"block_height\") VALUES(?,?,?,?,?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:42.33 UTC] INSERT INTO \"pool_production\"(\"pool_id\",\"slot\",\"header_hash\",\"parent_header_hash\",\"block_height\") VALUES(?,?,?,?,?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:42.33 UTC] INSERT INTO \"pool_production\"(\"pool_id\",\"slot\",\"header_hash\",\"parent_header_hash\",\"block_height\") VALUES(?,?,?,?,?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:42.33 UTC] INSERT INTO \"pool_production\"(\"pool_id\",\"slot\",\"header_hash\",\"parent_header_hash\",\"block_height\") VALUES(?,?,?,?,?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:42.33 UTC] INSERT INTO \"pool_production\"(\"pool_id\",\"slot\",\"header_hash\",\"parent_header_hash\",\"block_height\") VALUES(?,?,?,?,?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:42.33 UTC] INSERT INTO \"pool_production\"(\"pool_id\",\"slot\",\"header_hash\",\"parent_header_hash\",\"block_height\") VALUES(?,?,?,?,?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:42.33 UTC] INSERT INTO \"pool_production\"(\"pool_id\",\"slot\",\"header_hash\",\"parent_header_hash\",\"block_height\") VALUES(?,?,?,?,?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:42.33 UTC] INSERT INTO \"pool_production\"(\"pool_id\",\"slot\",\"header_hash\",\"parent_header_hash\",\"block_height\") VALUES(?,?,?,?,?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:42.33 UTC] INSERT INTO \"pool_production\"(\"pool_id\",\"slot\",\"header_hash\",\"parent_header_hash\",\"block_height\") VALUES(?,?,?,?,?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:42.33 UTC] INSERT INTO \"pool_production\"(\"pool_id\",\"slot\",\"header_hash\",\"parent_header_hash\",\"block_height\") VALUES(?,?,?,?,?)\n[iohk.cardano-wallet.stake-pools:Debug:ThreadId 37] [2019-12-04 09:42:42.33 UTC] In sync with the node.\n[iohk.cardano-wallet.stake-pools:Debug:ThreadId 37] [2019-12-04 09:42:42.83 UTC] In sync with the node.\n\ESC[34m[iohk.cardano-wallet.stake-pools:Info:ThreadId 37]\ESC[0m [2019-12-04 09:42:43.33 UTC] Applying blocks [481262.13 ... 481262.13]\n\ESC[34m[iohk.cardano-wallet.stake-pools:Info:ThreadId 37]\ESC[0m [2019-12-04 09:42:43.33 UTC] Writing stake-distribution for epoch 481262\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:43.33 UTC] DELETE FROM \"stake_distribution\" WHERE (\"epoch\"=?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:43.33 UTC] INSERT INTO \"stake_distribution\"(\"pool_id\",\"epoch\",\"stake\") VALUES (?,?,?),(?,?,?),(?,?,?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:43.34 UTC] INSERT INTO \"pool_production\"(\"pool_id\",\"slot\",\"header_hash\",\"parent_header_hash\",\"block_height\") VALUES(?,?,?,?,?)\n[iohk.cardano-wallet.stake-pools:Debug:ThreadId 37] [2019-12-04 09:42:43.34 UTC] In sync with the node.\n[iohk.cardano-wallet.stake-pools:Debug:ThreadId 37] [2019-12-04 09:42:43.84 UTC] In sync with the node.\n[iohk.cardano-wallet.stake-pools:Debug:ThreadId 37] [2019-12-04 09:42:44.34 UTC] In sync with the node.\n[iohk.cardano-wallet.stake-pools:Debug:ThreadId 37] [2019-12-04 09:42:44.85 UTC] In sync with the node.\n\ESC[34m[iohk.cardano-wallet.stake-pools:Info:ThreadId 37]\ESC[0m [2019-12-04 09:42:45.35 UTC] Applying blocks [481262.14 ... 481262.14]\n\ESC[34m[iohk.cardano-wallet.stake-pools:Info:ThreadId 37]\ESC[0m [2019-12-04 09:42:45.35 UTC] Writing stake-distribution for epoch 481262\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:45.35 UTC] DELETE FROM \"stake_distribution\" WHERE (\"epoch\"=?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:45.35 UTC] INSERT INTO \"stake_distribution\"(\"pool_id\",\"epoch\",\"stake\") VALUES (?,?,?),(?,?,?),(?,?,?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:45.35 UTC] INSERT INTO \"pool_production\"(\"pool_id\",\"slot\",\"header_hash\",\"parent_header_hash\",\"block_height\") VALUES(?,?,?,?,?)\n[iohk.cardano-wallet.stake-pools:Debug:ThreadId 37] [2019-12-04 09:42:45.36 UTC] In sync with the node.\n[iohk.cardano-wallet.stake-pools:Debug:ThreadId 37] [2019-12-04 09:42:45.86 UTC] In sync with the node.\n[iohk.cardano-wallet.stake-pools:Debug:ThreadId 37] [2019-12-04 09:42:46.36 UTC] In sync with the node.\n[iohk.cardano-wallet.stake-pools:Debug:ThreadId 37] [2019-12-04 09:42:46.86 UTC] In sync with the node.\n\ESC[34m[iohk.cardano-wallet.stake-pools:Info:ThreadId 37]\ESC[0m [2019-12-04 09:42:47.36 UTC] Applying blocks [481262.15 ... 481262.15]\n\ESC[34m[iohk.cardano-wallet.stake-pools:Info:ThreadId 37]\ESC[0m [2019-12-04 09:42:47.36 UTC] Writing stake-distribution for epoch 481262\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:47.36 UTC] DELETE FROM \"stake_distribution\" WHERE (\"epoch\"=?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:47.36 UTC] INSERT INTO \"stake_distribution\"(\"pool_id\",\"epoch\",\"stake\") VALUES (?,?,?),(?,?,?),(?,?,?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:47.36 UTC] INSERT INTO \"pool_production\"(\"pool_id\",\"slot\",\"header_hash\",\"parent_header_hash\",\"block_height\") VALUES(?,?,?,?,?)\n[iohk.cardano-wallet.stake-pools:Debug:ThreadId 37] [2019-12-04 09:42:47.37 UTC] In sync with the node.\n[iohk.cardano-wallet.stake-pools:Debug:ThreadId 37] [2019-12-04 09:42:47.87 UTC] In sync with the node.\n[iohk.cardano-wallet.stake-pools:Debug:ThreadId 37] [2019-12-04 09:42:48.37 UTC] In sync with the node.\n[iohk.cardano-wallet.stake-pools:Debug:ThreadId 37] [2019-12-04 09:42:48.88 UTC] In sync with the node.\n\ESC[34m[iohk.cardano-wallet.stake-pools:Info:ThreadId 37]\ESC[0m [2019-12-04 09:42:49.38 UTC] Applying blocks [481262.16 ... 481262.16]\n\ESC[34m[iohk.cardano-wallet.stake-pools:Info:ThreadId 37]\ESC[0m [2019-12-04 09:42:49.38 UTC] Writing stake-distribution for epoch 481262\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:49.38 UTC] DELETE FROM \"stake_distribution\" WHERE (\"epoch\"=?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:49.38 UTC] INSERT INTO \"stake_distribution\"(\"pool_id\",\"epoch\",\"stake\") VALUES (?,?,?),(?,?,?),(?,?,?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:49.38 UTC] INSERT INTO \"pool_production\"(\"pool_id\",\"slot\",\"header_hash\",\"parent_header_hash\",\"block_height\") VALUES(?,?,?,?,?)\n[iohk.cardano-wallet.stake-pools:Debug:ThreadId 37] [2019-12-04 09:42:49.39 UTC] In sync with the node.\n[iohk.cardano-wallet.stake-pools:Debug:ThreadId 37] [2019-12-04 09:42:49.89 UTC] In sync with the node.\n[iohk.cardano-wallet.stake-pools:Debug:ThreadId 37] [2019-12-04 09:42:50.39 UTC] In sync with the node.\n[iohk.cardano-wallet.stake-pools:Debug:ThreadId 37] [2019-12-04 09:42:50.89 UTC] In sync with the node.\n\ESC[34m[iohk.cardano-wallet.stake-pools:Info:ThreadId 37]\ESC[0m [2019-12-04 09:42:51.39 UTC] Applying blocks [481262.17 ... 481262.17]\n\ESC[34m[iohk.cardano-wallet.stake-pools:Info:ThreadId 37]\ESC[0m [2019-12-04 09:42:51.39 UTC] Writing stake-distribution for epoch 481262\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:51.39 UTC] DELETE FROM \"stake_distribution\" WHERE (\"epoch\"=?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:51.39 UTC] INSERT INTO \"stake_distribution\"(\"pool_id\",\"epoch\",\"stake\") VALUES (?,?,?),(?,?,?),(?,?,?)\n[iohk.cardano-wallet.query:Debug:ThreadId 37] [2019-12-04 09:42:51.39 UTC] INSERT INTO \"pool_production\"(\"pool_id\",\"slot\",\"header_hash\",\"parent_header_hash\",\"block_height\") VALUES(?,?,?,?,?)\n[iohk.cardano-wallet.stake-pools:Debug:ThreadId 37] [2019-12-04 09:42:51.40 UTC] In sync with the node.\n[iohk.cardano-wallet.stake-pools:Debug:ThreadId 37] [2019-12-04 09:42:51.90 UTC] In sync with the node.\n" does not contain "Logging shutdown"

  To rerun use: --match "/CLI Specifications/LOGGING - cardano-wallet serve logging [SERIAL]/LOGGING - Serve shuts down logging correctly/"

Randomized with seed 1273040386

Finished in 1049.3688 seconds
1027 examples, 21 failures, 1 pending


```
Also `jormungandr.exe` "stays" there and keeps logging to console after the tests are finished...

# Failure / Counter-example

---

# Resolution 

 - #1125 - Fixes `TRANS_EXTERNAL_CREATE_01` and `CLI_VERSION`

---

# QA

**Comment by rvl** [comment]
Thanks

- The PermissionDenied errors should be easily fixed (windows file locking).
- The assertion for CLI_VERSION could be adjusted to only consider the version part.
- Maybe TRANS_EXTERNAL_CREATE_01 is a flaky test that also fails on Linux?
- The 4 other failures need more investigation.

It would be nice to find out which (failing) tests leave behind the `jormungandr.exe` process - it's probably the DaedalusIPC tests, but may be other launch tests.

**Comment by piotr-iohk** [comment]
**Edit: updated Failing/Passing after #1125**
Executing one by one I found that quite a few were passing, and none of those 21 did not leave hanging Jormungandr:

Failing :x: :
```
--match "/CLI Specifications/DaedalusIPC [SERIAL]/Regression test for #1036/"
--match "/CLI Specifications/LOGGING - cardano-wallet serve logging [SERIAL]/LOGGING - Serve shuts down logging correctly/"
--match "/CLI Specifications/DaedalusIPC [SERIAL]/Should reply with the port --port/"
--match "/CLI Specifications/DaedalusIPC [SERIAL]/Should reply with the port --random/"
--match "/No backend required/Cardano.Wallet.NetworkSpec/Error paths/networkTip: ErrNetworkUnreachable/"
--match "/No backend required/Cardano.Wallet.NetworkSpec/Error paths/nextBlocks: ErrNetworkUnreachable/"
--match "/No backend required/Cardano.Wallet.NetworkSpec/White-box error path tests/getBlock/invalid url path/"
--match "/No backend required/Cardano.Wallet.NetworkSpec/White-box error path tests/getDescendantIds/invalid url path/"
--match "/No backend required/Cardano.Wallet.NetworkSpec/White-box error path tests/getStakeDistribution/invalid url path/"

```

Passing :heavy_check_mark: :
```
--match "/API Specifications/TRANS_EXTERNAL_CREATE_01 - proper single output transaction and proper binary format/"
--match "/No backend required/Miscellaneous CLI tests/CLI_VERSION - cardano-wallet shows version/"
--match "/No backend required/Cardano.Wallet.NetworkSpec/Happy Paths/no blocks after the tip/"
--match "/No backend required/Cardano.Wallet.NetworkSpec/Happy Paths/initiates recovery when the intersection is unknown/"
--match "/No backend required/Cardano.Wallet.NetworkSpec/Happy Paths/getAccountState: successfully gets account state/"
--match "/No backend required/Cardano.Wallet.NetworkSpec/Happy Paths/get network tip/"
--match "/No backend required/Cardano.Wallet.NetworkSpec/Happy Paths/get some blocks from the genesis/"
--match "/No backend required/Cardano.Wallet.NetworkSpec/Error paths/networkTip: throws on invalid url/"
--match "/No backend required/Cardano.Wallet.NetworkSpec/White-box error path tests/getAccountState/resource not found/"
--match "/No backend required/Cardano.Wallet.NetworkSpec/White-box error path tests/getAccountState/invalid url path/"
--match "/No backend required/Cardano.Wallet.NetworkSpec/White-box error path tests/getBlock/resource not found/"
--match "/No backend required/Cardano.Wallet.NetworkSpec/Submitting signed transactions (that are not obviously wrong)/empty tx succeeds/"
```
Perhaps those that are now passing were failing due to this unexpected Jormungandr hanging in bg...

**Comment by KtorZ** [comment]
That is very plausible @piotr-iohk :thinking:

**Comment by rvl** [comment]
I had a closer look at the `removeDirectoryRecursive` errors.

```
uncaught exception: IOException of type PermissionDenied
       C:\Users\katar\AppData\Local\Temp\cardano-wallet-jormungandr-dbc61946bb87b0ae\chain\blocks.sqlite-shm: removeDirectoryRecursive:removeContentsRecursive:removePathRecursive:removeContentsRecursive:removePathRecursive:DeleteFile "\\\\?\\C:\\Users\\katar\\AppData\\Local\\Temp\\cardano-wallet-jormungandr-dbc61946bb87b0ae\\chain\\blocks.sqlite-shm": permission denied (Odmowa dostępu.)
```

Actually it's not simply a matter of ignoring the exception, as I previously thought.

By the time `removeDirectoryRecursive` is called, the launch code should have ensured that `jormungandr.exe` has exited. But this is not the case.

**Comment by piotr-iohk** [comment]
Running tests on recent master (https://hydra.iohk.io/build/1552156) and there are only 2 failures:

```
Failures:

  test/integration/Cardano/Wallet/Jormungandr/NetworkSpec.hs:371:9:
  1) No backend required, Cardano.Wallet.NetworkSpec, Submitting signed transactions (that are not obviously wrong), no input, one output
       uncaught exception: IOException of type PermissionDenied
       C:\Users\piotr\AppData\Local\Temp\cardano-wallet-jormungandr-ee837d6b69462b5c\chain\blocks.sqlite-shm: removeDirectoryRecursive:removeContentsRecursive:removePathRecursive:removeContentsRecursive:removePathRecursive:DeleteFile "\\\\?\\C:\\Users\\piotr\\AppData\\Local\\Temp\\cardano-wallet-jormungandr-ee837d6b69462b5c\\chain\\blocks.sqlite-shm": permission denied (Odmowa dostępu.)

  To rerun use: --match "/No backend required/Cardano.Wallet.NetworkSpec/Submitting signed transactions (that are not obviously wrong)/no input, one output/"

  test/integration/Test/Integration/Jormungandr/Scenario/CLI/Server.hs:169:17:
  2) CLI Specifications, LOGGING - cardano-wallet serve logging [SERIAL], LOGGING - Serve shuts down logging correctly
       "... a lot of logs here..." does not contain "Logging shutdown"

  To rerun use: --match "/CLI Specifications/LOGGING - cardano-wallet serve logging [SERIAL]/LOGGING - Serve shuts down logging correctly/"

Randomized with seed 774828659

Finished in 2386.2869 seconds
1108 examples, 2 failures, 1 pending
```

The first one however passes when re-running with ` --match " --match "/No backend required/Cardano.Wallet.NetworkSpec/Submitting signed transactions (that are not obviously wrong)/no input, one output/""`
The second one probably can be fixed... I think...

**Comment by KtorZ** [comment]
@piotr-iohk are you also able to run all tests at once now or do you still run things one-by-one ?

**Comment by piotr-iohk** [comment]
@KtorZ I've run everything at once.

**Comment by piotr-iohk** [comment]
Today, after the fix - https://github.com/input-output-hk/cardano-wallet/pull/1255 - I get such errors on integration tests:

```
Failures:

  test/integration/Cardano/Wallet/Jormungandr/NetworkSpec.hs:418:5:
  1) No backend required, Cardano.Wallet.NetworkSpec, White-box error path tests, getAccountState, invalid url path
       uncaught exception: IOException of type PermissionDenied
       C:\Users\piotr\AppData\Local\Temp\cardano-wallet-jormungandr-2b1b7f7322ffdb56\chain\blocks.sqlite-shm: removeDirectoryRecursive:removeContentsRecursive:removePathRecursive:removeContentsRecursive:removePathRecursive:DeleteFile "\\\\?\\C:\\Users\\piotr\\AppData\\Local\\Temp\\cardano-wallet-jormungandr-2b1b7f7322ffdb56\\chain\\blocks.sqlite-shm": permission denied (Odmowa dostępu.)

  To rerun use: --match "/No backend required/Cardano.Wallet.NetworkSpec/White-box error path tests/getAccountState/invalid url path/"

  test/integration/Cardano/Wallet/Jormungandr/NetworkSpec.hs:433:5:
  2) No backend required, Cardano.Wallet.NetworkSpec, White-box error path tests, getBlock, invalid resource id
       uncaught exception: IOException of type PermissionDenied
       C:\Users\piotr\AppData\Local\Temp\cardano-wallet-jormungandr-8b6338dae37ed1d9\chain\blocks.sqlite-shm: removeDirectoryRecursive:removeContentsRecursive:removePathRecursive:removeContentsRecursive:removePathRecursive:DeleteFile "\\\\?\\C:\\Users\\piotr\\AppData\\Local\\Temp\\cardano-wallet-jormungandr-8b6338dae37ed1d9\\chain\\blocks.sqlite-shm": permission denied (Odmowa dostępu.)

  To rerun use: --match "/No backend required/Cardano.Wallet.NetworkSpec/White-box error path tests/getBlock/invalid resource id/"

Randomized with seed 789715958

Finished in 2767.1534 seconds
1107 examples, 2 failures, 1 pending
```

However, the other day, I think all tests passed for me. :thinking: 

Another issue is that after tests are finished - Jormungandr is not killed but left working and logging to the console.

**Comment by piotr-iohk** [comment]
On the latest https://hydra.iohk.io/build/1566747 all integration tests are passing when running _standalone_:

```
Finished in 2690.9991 seconds
1107 examples, 0 failures, 1 pending
```

Also unit tests passed for me for core and jormungandr...

**Comment by KtorZ** [comment]
@piotr-iohk Shall we close this one then? Perhaps we can now open a new ticket to replace this one about getting the tests to run properly in Hydra? (cc @rvl)

**Comment by piotr-iohk** [comment]
Closing. New issue created for tracking failing tests on Hydra -> https://github.com/input-output-hk/cardano-wallet/issues/1283.
